src/scripting.c

```diff
@@ -748,7 +748,7 @@ int luaRedisGenericCommand(lua_State *lua, int raise_error) {
     /* Convert the result of the Redis command into a suitable Lua type.
      * The first thing we need is to create a single string from the client
      * output buffers. */
-    if (listLength(c->reply) == 0 && c->bufpos < PROTO_REPLY_CHUNK_BYTES) {
+    if (listLength(c->reply) == 0 && c->bufpos < (long)c->buf_usable_size) {
```

Comment:
The `(long)` cast here it a bit unexpected. Since `c->bufpos` is an `int` I'd expect an `int` cast. But actually the correct thing to do is change `c->bufpos` to a `size_t` and avoid the cast altogether.

Reply:
OK, thanks